test/e2e/autoscale_test.go

```diff
@@ -0,0 +1,284 @@
+		if ctx != nil {
+			ctx.Cleanup()
+		}
+		require.FailNow(t, "Failed in prepare")
+	}
+	fw = framework.Global
+	namespace = ctx.GetID()
+	require.NotNil(t, namespace, "GetID failed")
+
+	addToFrameworkSchemeForSmokeTests(t)
+
+	if isOpenShift(t) {
+		esServerUrls = "http://elasticsearch." + storageNamespace + ".svc.cluster.local:9200"
+	}
+}
+
+func (suite *AutoscaleTestSuite) TearDownSuite() {
+	handleSuiteTearDown()
+}
+
+func TestAutoscaleSuite(t *testing.T) {
+	suite.Run(t, new(AutoscaleTestSuite))
+}
+
+func (suite *AutoscaleTestSuite) SetupTest() {
+	t = suite.T()
+}
+
+func (suite *AutoscaleTestSuite) AfterTest(suiteName, testName string) {
+	handleTestFailure()
+}
+
+func (suite *AutoscaleTestSuite) TestAutoScaleCollector() {
+	if !isOpenShift(t) {
+		t.Skip("This test is currently only supported on OpenShift")
+	}
+	waitForElasticSearch()
+
+	jaegerInstanceName := "simple-prod"
+	exampleJaeger := getSimpleProd(jaegerInstanceName, namespace, cpuResourceLimit, memoryResourceLimit, true)
+	err := fw.Client.Create(context.TODO(), exampleJaeger, &framework.CleanupOptions{TestContext: ctx, Timeout: timeout, RetryInterval: retryInterval})
+	require.NoError(t, err, "Error deploying example Jaeger")
+	defer undeployJaegerInstance(exampleJaeger)
+
+	err = e2eutil.WaitForDeployment(t, fw.KubeClient, namespace, jaegerInstanceName+"-collector", 1, retryInterval, timeout)
+	require.NoError(t, err, "Error waiting for collector deployment")
+
+	err = e2eutil.WaitForDeployment(t, fw.KubeClient, namespace, jaegerInstanceName+"-query", 1, retryInterval, timeout)
+	require.NoError(t, err, "Error waiting for query deployment")
+	logrus.Infof("Jaeger deploy finished, deploying tracegen in %s", namespace)
+
+	tracegen := createTracegenDeployment(jaegerInstanceName, namespace, "collector-headless", tracegenDurationInMinutes, replicas)
+	defer deleteTracegenDeployment(tracegen)
+
+	maxCollectorCount := -1
+	colletorPodListOptions := metav1.ListOptions{
+		LabelSelector: "app.kubernetes.io/name=" + jaegerInstanceName + "-collector",
+	}
+	lastIterationTimestamp := time.Now()
+	for i := 1; i <= tracegenDurationInMinutes; i++ {
+		pods, err := fw.KubeClient.CoreV1().Pods(namespace).List(context.Background(), colletorPodListOptions)
+		require.NoError(t, err)
+
+		collectorCount := len(pods.Items)
+		logrus.Infof("Iteration %d found %d pods", i, collectorCount)
+		if collectorCount > maxCollectorCount {
+			maxCollectorCount = collectorCount
+			if quitOnFirstScale && i > 1 {
+				break
+			}
+		}
+
+		// Print events since last iteration.
+		eventList, err := fw.KubeClient.CoreV1().Events(namespace).List(context.Background(), metav1.ListOptions{})
+		require.NoError(t, err)
+		for _, event := range eventList.Items {
+			if event.LastTimestamp.After(lastIterationTimestamp) {
+				logrus.Warnf("Event Type: %s Reason: %s Message: %s Time %v", event.Type, event.Reason, event.Message, event.LastTimestamp)
+			}
+		}
+
+		lastIterationTimestamp = time.Now()
```

Comment:
Might be better to set the `lastIterationTimestamp` to the last item's (in the event list) timestemp - I assume they are time ordered? Only setting to `time.Now` if the list is empty.